src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateIndexCommand.java

```diff
@@ -0,0 +1,94 @@
+@Schema(description = "Command that creates an index for a column in a table.")
+@JsonTypeName("createIndex")
+public record CreateIndexCommand(
+    @NotNull
+        @Size(min = 1, max = 48)
+        @Pattern(regexp = "[a-zA-Z][a-zA-Z0-9_]*")
+        @Schema(description = "Name of the column to create the index on")
+        String name,
+    @NotNull
+        @Schema(
+            description = "Definition for created index for a column.",
+            type = SchemaType.OBJECT)
+        Definition definition,
+    @Nullable @Schema(description = "Creating index command option.", type = SchemaType.OBJECT)
+        Options options)
+    implements CollectionCommand {
+  public record Definition(
+      @NotNull
+          @Size(min = 1, max = 48)
+          @Pattern(regexp = "[a-zA-Z][a-zA-Z0-9_]*")
+          @Schema(description = "Name of the column for which index to be created.")
+          String column,
+      @Nullable @Schema(description = "Different indexing options.", type = SchemaType.OBJECT)
+          Options options) {
+    // This is index definition options for text and vector column types.
+    public record Options(
+        @Nullable
+            @Schema(
+                description = "Ignore case in matching string values.",
+                defaultValue = "true",
+                type = SchemaType.BOOLEAN,
+                implementation = Boolean.class)
+            Boolean caseSensitive,
+        @Nullable
+            @Schema(
+                description = "When set to true, perform Unicode normalization on indexed strings.",
+                defaultValue = "false",
+                type = SchemaType.BOOLEAN,
+                implementation = Boolean.class)
+            Boolean normalize,
+        @Nullable
+            @Schema(
+                description =
+                    "When set to true, index will converts alphabetic, numeric, and symbolic characters to the ascii equivalent, if one exists.",
+                defaultValue = "false",
+                type = SchemaType.BOOLEAN,
+                implementation = Boolean.class)
+            Boolean ascii,
+        @Nullable
+            @Pattern(
+                regexp = "(dot_product|cosine|euclidean)",
+                message = "function name can only be 'dot_product', 'cosine' or 'euclidean'")
+            @Schema(
+                description =
+                    "Similarity function algorithm that needs to be used for vector search",
+                defaultValue = "cosine",
+                type = SchemaType.STRING,
+                implementation = String.class)
+            SimilarityFunction metric,
+        @Nullable
+            @Size(min = 1, max = 48)
+            @Pattern(regexp = "[a-zA-Z][a-zA-Z0-9_]*")
+            @Schema(description = "Model name used to generate the embeddings.")
+            String sourceModel) {}
+  }
+
+  // This is index command option irrespective of column definition.
+  public record Options(
```

Comment:
It seems bit confusing to me to have two Record types with same name (`Options`) here.
They are namespaced differently so JVM/JDK don't care, but to a reader it seems that maybe it'd be good to name the, differently. Not sure what'd be good names -- ColumnOptions vs IndexOptions?

Also: did we decide to have just one same "AddIndex" command for all different types?

Reply:
Agreed, they are bit confusing.  Not sure how to distinguish between Options at command level (for if not exist) and at column definition level.
Did we conclude having it multiple commands, I can split it into 2 commands. But will be repetition of the same code since we decided to put vectorize part of createTable..